HPXMLtoOpenStudio/resources/vehicle.rb

```diff
@@ -0,0 +1,223 @@
+        discharging_schedule = discharging_schedule.schedule
+      end
+    else
+      runner.registerWarning("Both schedule file and weekday fractions provided for '#{charging_col}' and '#{discharging_col}'; weekday fractions will be ignored.") if !vehicle.ev_charging_weekday_fractions.nil?
+      runner.registerWarning("Both schedule file and weekend fractions provided for '#{charging_col}' and '#{discharging_col}'; weekend fractions will be ignored.") if !vehicle.ev_charging_weekend_fractions.nil?
+      runner.registerWarning("Both schedule file and monthly multipliers provided for '#{charging_col}' and '#{discharging_col}'; monthly multipliers will be ignored.") if !vehicle.ev_charging_monthly_multipliers.nil?
+    end
+
+    return charging_schedule, discharging_schedule
+  end
+
+  # Apply an electric vehicle to the model using the battery.rb Battery class, which assigns OpenStudio ElectricLoadCenterStorageLiIonNMCBattery and ElectricLoadCenterDistribution objects.
+  # An EMS program models the effect of ambient temperature on the effective power output.
+  # An EMS program writes a 'discharge offset' variable to omit this from aggregate home electricity outputs.
+  # If no charging/discharging schedule is provided, then the electric vehicle is not modeled.
+  # Bi-directional charging is not currently implemented
+  #
+  # @param runner [OpenStudio::Measure::OSRunner] Object typically used to display warnings
+  # @param model [OpenStudio::Model::Model] OpenStudio Model object
+  # @param spaces [Hash] Map of HPXML locations => OpenStudio Space objects
+  # @param hpxml_bldg [HPXML::Building] HPXML Building object representing an individual dwelling unit
+  # @param vehicle [HPXML::Vehicle] Object that defines a single electric vehicle
+  # @param schedules_file [SchedulesFile] SchedulesFile wrapper class instance of detailed schedule files
+  # @return [nil]
+  def self.apply_electric_vehicle(runner, model, spaces, hpxml_bldg, vehicle, schedules_file)
+    model.getElectricEquipments.sort.each do |ee|
+      if ee.endUseSubcategory.start_with? Constants::ObjectTypeMiscElectricVehicleCharging
+        runner.registerWarning('Electric vehicle was specified as a plug load and as a battery, vehicle charging will be modeled as a plug load.')
+        return
+      end
+    end
+
+    # Assign charging and vehicle space
+    ev_charger = vehicle.ev_charger
+    if ev_charger.nil?
+      runner.registerWarning('Electric vehicle specified with no charger provided; battery will not be modeled.')
+      return
+    end
+    vehicle.location = ev_charger.location
+
+    # Get schedules to calculate effective discharge power
+    charging_schedule, discharging_schedule = get_ev_charging_schedules(runner, model, vehicle, schedules_file)
+    if charging_schedule.nil? && discharging_schedule.nil?
+      runner.registerWarning('Electric vehicle battery specified with no charging/discharging schedule provided; battery will not be modeled.')
```

Comment:
I'm confused. Are schedules required? The schematron/docs say they are optional and that there are defaults.

Reply:
They are not, removed this in https://github.com/NREL/OpenStudio-HPXML/pull/1533/commits/97450dca0c343296c903278bacdf310b21ef6699